File: test/array/design-hit-counter.test.ts
import { HitCounter } from '../../src/array/design-hit-counter';

describe('design hit counter', () => {
  test('design hit counter - test case 1', async () => {
    const counter = new HitCounter();

    counter.hit(1);
    counter.hit(2);
    counter.hit(3);

    expect(counter.getHits(4)).toBe(3);

    counter.hit(300);

    expect(counter.getHits(300)).toBe(4);
    expect(counter.getHits(301)).toBe(3);
  });
});

File: test/leetcode/array/design_hit_counter_test.py
from leetcode.array.design_hit_counter import HitCounter


hc = HitCounter()


def test_case_1():
    hc.hit(1)
    hc.hit(2)
    hc.hit(3)

    assert hc.getHits(4) == 3

    hc.hit(300)

    assert hc.getHits(300) == 4
    assert hc.getHits(301) == 3
